fragalysis_api/xcimporter/conversion_pdb_mol.py

```diff
@@ -275,15 +269,21 @@ def create_metadata_file(self, directory, file_base, smiles_file=None):
             except Exception as e:
                 print(e)
                 print('failed to open smiles file ' + smiles_file)
-                smiles = 'NA'
 
         else:
             print(f'Warning: No smiles file: {file_base}')
-            smiles = 'NA'
+            # Try use mol_obj if no smiles file
+            try:
+                smiles = Chem.MolToSmiles(mol_obj)
+            except Exception as e:
+                print(e)
+                print('failed to convert mol obj to smiles' + smiles_file)
+
+                smiles = "NA"
 
         meta_data_dict = {'Blank':'',
                           'fragalysis_name': file_base,
-                          'crystal_name': crystal_name,
+                          'crystal_name': file_base.split("_")[0],
```

Comment:
what if the crystal name has an underscore in it?

Reply:
yes that will be an issue - as per your suggestion keep underscores out or probably better convert files with underscores to hyphens.

Reply:
catching everything before the final underscore has done the trick